server/services/google-cast/lib/google_cast.setValue.js

```diff
@@ -22,6 +23,16 @@ async function setValue(device, deviceFeature, value) {
     client.connect(ipAddress, () => {
       logger.debug('Google Cast Connected, launching app ...');
 
+      if (options.volume) {
+        client.setVolume({ level: options.volume / 100 }, (err, newvol) => {
+          if (err) {
+            logger.debug('there was an error setting the volume');
+          } else {
+            logger.debug('volume changed to %s', newvol);
+          }
+        });
+      }
+
```

Comment:
You're not waiting for the callback to finish, are you sure the volume will be changed when client.launch will be started ?

You can use promisify in utils to turn this in a promise for example

Reply:
Done 👍 . Volume is set immediatly, when we play audio from url it needs time to load it, so I thought is was not necessary